### PR TITLE
[Catalog] Add catalog schema.

### PR DESCRIPTION
**Pull request by MUzairS15**
**Notes for Reviewers**

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: schemas/constructs/v1beta1/catalog_data.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            "description": "Categorization of the type of design or operational flow depicted in this design."
+        },
+        "snapshotURL": {
+            "type": "array",
+            "items": {
+                "type": "string",
+                "format": "uri",
+                "pattern": "^(https?|http?|oci)://"
+            },
+            "description": "Contains reference to the dark and light mode snapshots of the catalog."
```
Comment (MUzairS15) [inline schemas/constructs/v1beta1/catalog_data.json:51]:
```suggestion
            "description": "Contains reference to the dark and light mode snapshots of the design."
```

File: schemas/constructs/v1beta1/catalog_data.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "properties": {
+        "compatibility": {
+            "type": "array",
+            "title": "Technology",
+            "items": {
+                "enum": [
+                    "kubernetes"
```
Comment (leecalcote) [inline schemas/constructs/v1beta1/catalog_data.json:10]: $ref: ../../model.schema

File: schemas/constructs/v1beta1/catalog_data.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                ],
+                "type": "string"
+            },
+            "uniqueItems": true,
+            "minItems": 1,
```
Comment (leecalcote) [inline schemas/constructs/v1beta1/catalog_data.json:15]:
```suggestion
            "$comment": "Content published into the catalog must reference at least one model, and by implication, contain at least one component.",
            "minItems": 1,
```

File: schemas/constructs/v1beta1/catalog_data.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            "description": "A list of technologies included in or implicated by this design; a list of relevant technology tags."
```
Comment (leecalcote) [inline schemas/constructs/v1beta1/catalog_data.json:16]:
```suggestion
            "description": "A list of models implicated by components in this catalog item (i.e. design)."
```

File: schemas/constructs/v1beta1/catalog_data.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "properties": {
+        "compatibility": {
+            "type": "array",
+            "title": "Technology",
```
Comment (leecalcote) [inline schemas/constructs/v1beta1/catalog_data.json:7]:
```suggestion
            "title": "Model",
```

**Pull request merged**